samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -141,15 +142,36 @@ class TaskInstance(
     collector.register
   }
 
-  def registerConsumers {
+  /**
+    * Computes the starting offset for the partitions assigned to the task and registers them with the underlying {@see SystemConsumers}.
+    *
+    * Starting offset for a partition of the task is computed in the following manner:
+    *
+    * 1. If a startpoint exists for a task, system stream partition and it resolves to a offset, then the resolved offset is used as the starting offset.
+    * 2. Else, the checkpointed offset for the system stream partition is used as the starting offset.
+    */
+  def registerConsumers() {
     debug("Registering consumers for taskName: %s" format taskName)
     systemStreamPartitions.foreach(systemStreamPartition => {
-      val startingOffset = getStartingOffset(systemStreamPartition)
-      val startpoint = offsetManager.getStartpoint(taskName, systemStreamPartition).getOrElse(null)
-      consumerMultiplexer.register(systemStreamPartition, startingOffset, startpoint)
-      metrics.addOffsetGauge(systemStreamPartition, () =>
-          offsetManager.getLastProcessedOffset(taskName, systemStreamPartition).orNull
-        )
+      var startingOffset: String = getStartingOffset(systemStreamPartition)
+      val startpoint: Option[Startpoint] = offsetManager.getStartpoint(taskName, systemStreamPartition)
+      if (startpoint != null && startpoint.isDefined) {
```

Comment:
nit: Probably cleaner to use scala `match`:
```
startpoint match {
  case Some(sp) => ...
  case None => ...
}
```

Reply:
Done.